pkg/protocols/http/http.go

```diff
@@ -388,8 +390,13 @@ func (request *Request) Compile(options *protocols.ExecutorOptions) error {
 		}
 	}
 	if len(request.Payloads) > 0 {
+		// specifically for http requests high concurrency and and threads will lead to memory exausthion, hence reduce the maximum parallelism
+		if protocolstate.MemGuardian.Warning.Load() {
+			request.Threads = 5
```

Comment:
can this be configurable as a setting?

Reply:
Definitely - I'm going to refactor it and allow configuration via environment variable